addons/account/models/account_payment.py

```diff
@@ -358,6 +356,23 @@ def _prepare_move_line_default_vals(self, write_off_line_vals=None):
     # COMPUTE METHODS
     # -------------------------------------------------------------------------
 
+    @api.depends('company_id', 'move_id.journal_id')
+    def _compute_journal_id(self):
+        self.move_id.filtered(
+            lambda move: move.journal_id.type not in ('bank', 'cash')
+        ).with_context(is_payment=True)._compute_journal_id()
+        for record in self:
+            record.journal_id = record.move_id.with_context(is_payment=True).journal_id
+
+    @api.onchange('journal_id')
+    def _inverse_journal_id(self):
+        for record in self:
+            record.move_id.name = False
+            record.move_id.journal_id = record.journal_id
+
+    def _search_payment_id(self, operator, value):
+        return [('move_id.journal_id', operator, value)]
+
```

Comment:
It is more correct and robust (doesn't only work with `onchange`/`default_get`) and you don't have to "hack" the `create` to compute values :shrug:

Reply:
Solved with
```python
    def new(self, values=None, origin=None, ref=None):
        self = self.with_context(is_payment=True)
        payment = super().new(values, origin, ref)
        if not payment.journal_id:  # might not be computed because declared by inheritance
            payment.move_id._compute_journal_id()
        return payment
```

IMO, this needs to be done by the ORM though